### PR TITLE
Restructure Codebase and Update workflow

### PR DESCRIPTION
**Pull request by koustuvsinha**
Major restructure of codebase and updated workflow to be easier to understand and work on.

### CODE REVIEW

File: lib/token.js [hunk 1]
```diff
@@ -7,7 +7,8 @@ module.exports.generateToken = function () {
 
 	try {
 		crypto = require('crypto')
-		var secret = require('../config/config').secret
+		//var secret = require('../config/config').secret
+		var secret = process.env.RPSECRET
```
Comment (alayek) [inline lib/token.js:11]: How did you get this back? I thought the encrypted env var support is not there.

File: test/repl_test.js [hunk 2]
```diff
@@ -13,6 +13,8 @@ var colors = require('colors')
 var payload = ''
 var count = 0
 
+var PADDING = 19
```
Comment (alayek) [inline test/repl_test.js:16]: Why this specific number 19?

File: lib/token.js (hunk 1 above)
Comment (koustuvsinha) [inline lib/token.js:11]: encrypted env var support is in the branches of this repo remember? :wink:

File: test/repl_test.js (hunk 2 above)
Comment (koustuvsinha) [inline test/repl_test.js:16]: in a test file, this meta info should be kept at the end : 

```
## $$TEST_METHOD$$
## test_hello_world()
```

So, from beginning $ to the space before `test_hello_world`, there are 19 characters, which serves as a padding to extract the method name.

The method name is required to run the tests in REPL.

File: lib/token.js (hunk 1 above)
Comment (alayek) [inline lib/token.js:11]: Right, forgot :sweat_smile:

File: test/repl_test.js (hunk 2 above)
Comment (alayek) [inline test/repl_test.js:16]: Where can I learn these stuff :open_mouth:

**Comment by alayek** [comment]
# LGTM :+1:

Let's :shipit: :package:

File: test/repl_test.js (hunk 2 above)
Comment (koustuvsinha) [inline test/repl_test.js:16]: this is basically how gulp injects includes and links in a html file, with the help of placeholders :stuck_out_tongue:

**Pull request merged**